/* ==========================================================
 *
 * Generated by system . DO NOT EDIT!
 * ========================================================== */


// gsk_pb_clt_data;
//-----------------------------好友列表
message pb_clt_t_friend
{
    optional int32                f_update_type                   = 1;  //--1:全量覆盖2:追加
    repeated pb_user_t_friend     f_friends                       = 2;  //--好友列表
}

//-----------------------------好友推荐列表
message pb_clt_t_recommend_friend
{
    optional int32                f_update_type                   = 1;  //--1:全量覆盖2:追加
    repeated pb_user_t_recommend_friend f_recommend_friends             = 2;  //--推荐列表
}

//-----------------------------群组中的成员
message pb_clt_t_group_user
{
    optional uint32               f_uin                           = 1;  //--用户id
    optional string               f_name                          = 2;  //--用户名称
    optional string               f_remark                        = 3;  //--用户在群组中的群昵称
    optional int32                f_title                         = 4;  //--用户在群组中身份
    optional string               f_initial                       = 5;  //--首字母
    optional int32                f_weight                        = 6;  //--权重
    optional string               f_name_pinyin                   = 7;  //--名称全拼
    optional string               f_name_pinyin_j                 = 8;  //--名称简拼
}

//-----------------------------单个聊天群组
message pb_clt_t_group_item
{
    optional string               f_group_id                      = 1;  //--群组id
    optional string               f_group_name                    = 2;  //--群组name
    optional string               f_group_desc                    = 3;  //--群组描述
    optional int32                f_group_type                    = 4;  //--群组类型
    optional int32                f_seq                           = 5;  //--群组消息当前seq
    optional string               f_last_msg                      = 6;  //--最后一条消息
    optional int32                f_last_msg_time                 = 7;  //--最后一条消息发送时间
    optional int32                f_last_msg_type                 = 8;  //--最后一条消息发送类型
    optional int32                f_last_sender_id                = 9;  //--最后一条消息发送者id
    optional string               f_last_sender_name              = 10;  //--最后一条消息发送者name
    repeated string               f_avatar                        = 11;  //--群组头像列表
    optional int32                f_is_screen                     = 12;  //--是否屏蔽群消息
    optional int32                f_is_remind                     = 13;  //--是否提醒消息
    optional string               f_remark                        = 14;  //--我在该群中的备注
    optional string               f_initial                       = 15;  //--首字母
    optional int32                f_weight                        = 16;  //--权重
    optional string               f_name_pinyin                   = 17;  //--群组名称全拼
    optional string               f_name_pinyin_j                 = 18;  //--群组名称简拼
    optional int32                f_member_count                  = 19;  //--群组成员数量
}

message pb_clt_t_group_details
{
    optional pb_clt_t_group_item  f_group_info                    = 1;  //--群组基础信息
    repeated pb_clt_t_group_user  f_members                       = 2;  //--群组成员
}

//-----------------------------聊天群组
message pb_clt_t_group
{
    optional int32                f_update_type                   = 1;  //--1:全量覆盖2:追加
    optional string               f_last_update_gid               = 2;  //--最后变化的群组id
    repeated pb_clt_t_group_item  f_im_group                      = 3;  //--群组列表
}

//-----------------------------版本更新
message pb_clt_t_version
{
    optional int32                f_version_id                    = 1;  //--版本id
    optional string               f_version_msg                   = 2;  //--版本信息
}

//-----------------------------返回给client的pb包
message pb_clt_t_user
{
    optional pb_user_t_info       f_info                          = 1;  //--用户profile
    optional pb_clt_t_friend      f_friend_list                   = 2;  //--好友列表
    optional pb_clt_t_friend      f_search_friend_list            = 3;  //--搜索好友列表
    optional pb_clt_t_friend      f_validation_friend_list        = 4;  //--验证好友列表
    optional pb_clt_t_friend      f_recommend_friend_list         = 5;  //--推荐好友列表
    optional pb_clt_t_group       f_im_group                      = 6;  //--聊天群组
    optional pb_user_t_info       f_friend_info                   = 7;  //--好友profile
    optional pb_im_t_msg          f_im_msg                        = 8;  //--im聊天消息列表
    optional pb_clt_t_group_details f_im_group_details              = 9;  //--聊天群组详情
    optional pb_prj_t_prj         f_prj_list                      = 10;  //--项目列表
    optional pb_prj_t_details     f_prj_details                   = 11;  //--项目详情
    optional pb_clt_t_recommend_friend f_n_recommend_friend_list       = 12;  //--推荐好友
    optional pb_prj_t_recommend_prj f_recommend_prj_list            = 13;  //--推荐项目列表
    repeated pb_prj_t_user        f_prj_user                      = 14;  //--项目成员
    repeated pb_prj_t_user        f_prj_join_user                 = 15;  //--申请加入项目成员
    optional pb_prj_t_over_view   f_prj_over_view                 = 16;  //--项目概览
    optional pb_clt_t_version     f_version                       = 17;  //--版本信息
}

// gsk_pb_im_data;
//-----------------------------群附件
message pb_im_t_msg_attach
{
    optional string               f_attach_name                   = 1;  //--附件名称
    optional int32                f_attach_type                   = 2;  //--附件类型
    optional string               f_attach_url                    = 3;  //--附件地址
    optional int32                f_attach_size                   = 4;  //--附件大小
    optional int32                f_attach_playtime               = 5;  //--附件播放时长
    optional int32                f_attach_uploadtime             = 6;  //--附件上传时间
}

//-----------------------------任务信息   
//-----------------------------”XXX发布一个任务“等文字描述放在单条聊天信息f_msg_body中
//-----------------------------f_operate_type 0：其他(不需要确认的通知)；1：通知任务发起者即将到期提醒接收者；2：通知任务接收者即将到期；
//-----------------------------3：通知任务发起者提醒接收者查看；4：通知任务接收者查看
message pb_im_t_msg_task
{
    optional string               f_task_id                       = 1;  //--任务id
    optional string               f_prj_id                        = 2;  //--项目id
    optional int32                f_task_type                     = 3;  //--任务类型
    optional string               f_task_desc                     = 4;  //--任务描述
    optional int32                f_task_desc_type                = 5;  //--任务描述类型；1：文字，2：语音
    optional string               f_position                      = 6;  //--部位
    repeated int32                f_receiver_uin                  = 7;  //--消息接收者uin
    optional int32                f_operate_type                  = 8;  //--消息通知类型
    optional string               f_assign_desc                   = 9;  //--任务指派描述
    optional int32                f_task_end                      = 10;  //--任务结束时间
}

//-----------------------------单条聊天信息
message pb_im_t_msg_item
{
    optional string               f_msg_id                        = 1;  //--消息id
    optional uint32               f_msg_type                      = 2;  //--消息类型/文本、语音...
    optional string               f_msg_body                      = 3;  //--消息内容
    optional int32                f_sender_id                     = 4;  //--消息发送者id
    optional string               f_sender_name                   = 5;  //--消息发送者name
    optional string               f_sender_remark                 = 6;  //--消息发送者在该群中的昵称
    optional string               f_sender_avatar                 = 7;  //--消息发送者头像
    optional string               f_group_id                      = 8;  //--群组id
    optional uint32               f_group_type                    = 9;  //--群组类型
    optional string               f_group_name                    = 10;  //--群组name
    optional string               f_group_desc                    = 11;  //--群组描述
    optional int32                f_send_time                     = 12;  //--发送时间
    optional int32                f_msg_seq_id                    = 13;  //--消息seq
    repeated pb_im_t_msg_attach   f_attch                         = 14;  //--附件
    optional pb_im_t_msg_task     f_msg_task                      = 15;  //--任务信息
}

//-----------------------------f_group_id中聊天列表
message pb_im_t_msg
{
    optional int32                f_update_type                   = 1;  //--1：全量覆盖
    optional string               f_group_id                      = 2;  //--聊天群组id
    repeated pb_im_t_msg_item     f_msg                           = 3;  //--该群组中的消息list
}

// gsk_pb_prj_data;
//-----------------------------项目中通讯录用户
message pb_prj_t_user
{
    optional int32                f_uid                           = 1;  //--用户id
    optional string               f_name                          = 2;  //--成员名称
    optional string               f_phone                         = 3;  //--成员电话
    optional string               f_job_type                      = 4;  //--岗位
    optional int32                f_add_time                      = 5;  //--添加时间
    optional int32                f_task_operate_status           = 6;  //--任务成员，任务操作状态；0:默认，1:已查看，.....
    optional string               f_join_content                  = 7;  //--加入项目理由
    optional string               f_name_pinyin                   = 8;  //--用户名称拼音
}

//-----------------------------项目FEED
message pb_prj_t_feed
{
    optional int32                f_time                          = 1;  //--feed产生时间
    optional string               f_msg                           = 2;  //--feed内容
}

//-----------------------------任务
message pb_prj_t_task
{
    optional string               f_task_id                       = 1;  //--任务id
    optional string               f_prj_id                        = 2;  //--项目id
    optional int32                f_task_type                     = 3;  //--任务类型
    optional string               f_task_desc                     = 4;  //--任务描述
    optional int32                f_task_desc_type                = 5;  //--任务描述类型；1：文字，2：语音
    optional pb_prj_t_user        f_sponsor                       = 6;  //--任务发起人
    repeated pb_prj_t_user        f_members                       = 7;  //--任务接收人
    optional int32                f_task_status                   = 8;  //--任务状态
    optional string               f_position                      = 9;  //--部位
    repeated pb_im_t_msg_attach   f_attach_list                   = 10;  //--附件列表
    optional int32                f_task_begin                    = 11;  //--任务开始时间
    optional int32                f_task_end                      = 12;  //--任务结束时间
}

//-----------------------------任务汇报
message pb_prj_t_task_report
{
    optional string               f_report_id                     = 1;  //--汇报id
    optional pb_prj_t_user        f_sponsor                       = 2;  //--汇报人
    optional pb_prj_t_user        f_reply_to                      = 3;  //--被回复人
    optional string               f_title                         = 4;  //--汇报标题
    optional string               f_content                       = 5;  //--汇报内容
    optional int32                f_content_type                  = 6;  //--汇报内容类型；1：文字，2：语音
    repeated pb_im_t_msg_attach   f_attach_list                   = 7;  //--附件列表
    optional int32                f_report_time                   = 8;  //--汇报时间
}

//-----------------------------任务查看信息
message pb_prj_t_task_view
{
    optional int32                f_uin                           = 1;  //--查看者id
    optional string               f_name                          = 2;  //--查看者昵称
    optional string               f_avatar                        = 3;  //--查看者头像
    optional int32                f_time                          = 4;  //--查看时间
}

//-----------------------------任务详情
message pb_prj_t_task_details
{
    optional pb_prj_t_task        f_task                          = 1;  //--任务信息
    repeated pb_prj_t_task_report f_report_list                   = 2;  //--汇报记录
    optional int32                f_report_count                  = 3;  //--汇报次数
    repeated pb_prj_t_task_view   f_view_list                     = 4;  //--查看记录
    optional int32                f_view_count                    = 5;  //--查看次数
    repeated string               f_position_list                 = 6;  //--常用部位列表
}

//-----------------------------项目基本信息
message pb_prj_t_info
{
    optional string               f_prj_id                        = 1;  //--项目id
    optional string               f_prj_name                      = 2;  //--项目名称
    optional int32                f_prj_type                      = 3;  //--项目类型
    optional string               f_prj_image                     = 4;  //--项目图片
    optional string               f_group_id                      = 5;  //--群组id
    optional string               f_province                      = 6;  //--项目省份
    optional string               f_city                          = 7;  //--项目城市
    optional string               f_prj_address                   = 8;  //--项目地址
    optional int32                f_member_count                  = 9;  //--项目成员数量
    optional int32                f_floor_area                    = 10;  //--建筑面积
    optional int32                f_prj_begin                     = 11;  //--项目开始日期
    optional int32                f_prj_end                       = 12;  //--项目结束日期
    optional string               f_jianzhu_danwei                = 13;  //--建筑单位
    optional string               f_shigong_danwei                = 14;  //--施工单位
    optional string               f_jianli_danwei                 = 15;  //--建设单位
    optional string               f_fenbao_danwei                 = 16;  //--分包单位
    optional double               f_lat                           = 17;  //--经纬度
    optional double               f_lon                           = 18;  //--经纬度
    optional string               f_job_type                      = 19;  //--岗位
    optional string               f_other_info                    = 20;  //--其他的一些信息
    optional string               f_sms_msg                       = 21;  //--短信通知信息
    repeated pb_prj_t_task        f_tasks                         = 22;  //--项目任务列表
    repeated pb_prj_t_task        f_owner_tasks                   = 23;  //--项目任务列表
    optional int32                f_owner_tasks_cnt               = 24;  //--项目结束日期
    optional int32                f_add_time                      = 25;  //--项目创建时间
    optional int32                f_creater_uin                   = 26;  //--项目创建者id
}

//-----------------------------项目日志
message pb_prj_t_log
{
    optional string               f_log_content                   = 1;  //--项目名称
    optional int32                f_add_time                      = 2;  //--项目二维码
}

//-----------------------------项目概览
message pb_prj_t_over_view
{
    optional string               f_prj_id                        = 1;  //--项目id
    optional string               f_prj_name                      = 2;  //--项目名称
    optional string               f_prj_code                      = 3;  //--项目二维码
    optional int32                f_prj_member_count              = 4;  //--项目成员总数
    optional string               f_prj_my_job                    = 5;  //--我的项目岗位
    optional int32                f_prj_begin                     = 6;  //--项目开始日期
    optional int32                f_prj_end                       = 7;  //--项目结束日期
    optional int32                f_prj_task_count                = 8;  //--任务总数
    repeated pb_prj_t_log         f_prj_log                       = 9;  //--项目日志列表
    optional pb_clt_t_friend      f_frd                           = 10;  //--项目申请加入成员
}

message pb_prj_t_prj
{
    optional int32                f_update_type                   = 1;  //--1:全量覆盖2:追加
    repeated pb_prj_t_info        f_prjs                          = 2;  //--项目列表
}

//-----------------------------推荐项目基本信息
message pb_prj_t_recommend_info
{
    optional pb_prj_t_info        f_info                          = 1;  //--项目基本信息
    optional uint32               f_type                          = 2;  //--推荐类型
    optional int32                f_weight                        = 3;  //--权重
    optional string               f_other_info                    = 4;  //--其他的一些信息
}

message pb_prj_t_recommend_prj
{
    optional int32                f_update_type                   = 1;  //--1:全量覆盖2:追加
    repeated pb_prj_t_recommend_info f_prjs                          = 2;  //--项目列表
}

//-----------------------------项目详情
message pb_prj_t_details
{
    optional pb_prj_t_info        f_info                          = 1;  //--项目基本信息
    optional pb_clt_t_group       f_im_group                      = 2;  //--项目群组
    optional pb_prj_t_task_details f_task_details                  = 3;  //--项目任务详情
    repeated pb_prj_t_task        f_tasks                         = 4;  //--项目任务列表
    optional int32                f_task_total_count              = 5;  //--项目任务总数
    optional int32                f_task_unfinished_count         = 6;  //--项目任务未完成数
    optional int32                f_task_delay_count              = 7;  //--项目任务延期数
    repeated pb_prj_t_user        f_users                         = 8;  //--项目通讯录用户列表
    repeated pb_prj_t_user        f_app_users                     = 9;  //--项目通讯录用户列表
    optional pb_clt_t_friend      f_frd                           = 10;  //--项目申请加入成员
    repeated pb_prj_t_feed        f_feeds                         = 11;  //--项目FEED
    optional pb_prj_t_over_view   f_prj_over_view                 = 12;  //--项目概览
}

// gsk_pb_user_data;
//-----------------------------用户PROFILE
message pb_user_t_info
{
    optional uint32               f_uin                           = 1;  //--用户id
    optional string               f_province                      = 2;  //--省份
    optional string               f_city                          = 3;  //--城市
    optional string               f_phone                         = 4;  //--手机号
    optional string               f_name                          = 5;  //--昵称
    optional string               f_company_type                  = 6;  //--公司类型
    optional string               f_company                       = 7;  //--公司类型
    optional string               f_years_of_working              = 8;  //--工作年限
    optional string               f_job_type                      = 9;  //--岗位
    optional string               f_job_title                     = 10;  //--职称
    optional int32                f_influence                     = 11;  //--影响力
    optional int32                f_setting                       = 12;  //--用户设置（bitmap方式）
    optional string               f_name_pinyin                   = 13;  //--昵称全拼
    optional string               f_name_pinyin_j                 = 14;  //--昵称简拼
}

//-----------------------------用户系统参数
message pb_user_t_extra
{
    optional int32                f_auto_inc_id                   = 1;  //--用户自增id
    optional int32                f_last_req_time                 = 2;  //--用户最后一次通信时间
}

//-----------------------------用户好友
message pb_user_t_friend
{
    optional uint32               f_uin                           = 1;  //--好友uin
    optional string               f_name                          = 2;  //--好友昵称
    optional string               f_remark                        = 3;  //--好友备注
    optional string               f_phone                         = 4;  //--手机号
    optional string               f_msg                           = 5;  //--消息
    optional int32                f_flag                          = 6;  //--好友状态
    optional string               f_group_id                      = 7;  //--聊天群组id
    optional string               f_initial                       = 8;  //--首字母
    optional int32                f_weight                        = 9;  //--权重
    optional string               f_name_pinyin                   = 10;  //--昵称全拼
    optional string               f_name_pinyin_j                 = 11;  //--昵称简拼
    optional string               f_remark_pinyin                 = 12;  //--备注全拼
    optional string               f_remark_pinyin_j               = 13;  //--备注简拼
}

//-----------------------------用户推荐好友
message pb_user_t_recommend_friend
{
    optional uint32               f_uin                           = 1;  //--好友uin
    optional string               f_name                          = 2;  //--好友昵称
    optional uint32               f_type                          = 3;  //--推荐类型
    optional int32                f_weight                        = 4;  //--权重
    optional string               f_name_pinyin                   = 5;  //--昵称全拼
    optional string               f_name_pinyin_j                 = 6;  //--昵称简拼
    optional string               f_company_type                  = 7;  //--公司类型
    optional string               f_company                       = 8;  //--公司名称
    optional string               f_other_info                    = 9;  //--显示信息
}

//-----------------------------用户上传的通讯录
message pb_user_t_phone_address
{
    optional string               f_name                          = 1;  //--好友昵称
    optional string               f_phone                         = 2;  //--好友手机号
}

//-----------------------------用户聊天群组
message pb_user_t_im_group
{
    optional string               f_group_id                      = 1;  //--聊天群组id
    optional int32                f_group_flag                    = 2;  //--聊天群组id
}

//-----------------------------用户项目列表
message pb_user_t_prj
{
    optional string               f_prj_id                        = 1;  //--项目id
    optional string               f_group_id                      = 2;  //--群组id
}

//-----------------------------用户存储结构
message pb_user_t_data
{
    optional pb_user_t_info       f_info                          = 1;  //--用户profile
    optional pb_user_t_extra      f_extra                         = 2;  //--用户额外参数
    repeated pb_user_t_friend     f_friend_list                   = 3;  //--用户好友
    repeated pb_user_t_friend     f_friend_validation_list        = 4;  //--验证列表
    repeated pb_user_t_friend     f_friend_recommend_list         = 5;  //--用户推荐
    repeated pb_user_t_im_group   f_im_group                      = 6;  //--我的群聊
    optional pb_user_t_info       f_other_info                    = 7;  //--好友profile
    repeated pb_user_t_prj        f_prj_list                      = 8;  //--我的项目
    repeated pb_user_t_recommend_friend f_recommend_friends             = 9;  //--推荐好友
    repeated pb_prj_t_recommend_info f_prjs                          = 10;  //--推荐项目列表
}

/* ==========================================================
 *
 * Generated by system . DO NOT EDIT!
 * ========================================================== */


